File: DevExpress-Demo/VB/PivotGridDemo.Wpf/Helpers/ServerMode/OrderDataContext.vb
Imports DevExpress.Xpf.DemoBase
Imports System.Data.Entity
Imports System.IO

Namespace PivotGridDemo

    Public Class OrderDataContext
        Inherits DbContext

        Public Shared ReadOnly FileName As String = Path.GetFullPath("OrderData.db")

        Public Sub New()
            MyBase.New(SQLiteDataBaseGenerator.CreateConnection(ServerModeViewModel.DatabaseInfo), True)
        End Sub

        Public Property OrderDataRecords As DbSet(Of OrderDataRecord)
    End Class

    Public Class OrderDataRecord

        Public Property Id As Integer

        Public Property OrderDate As Date

        Public Property Quantity As Integer

        Public Property UnitPrice As Decimal

        Public Property CustomerName As String

        Public Property ProductName As String

        Public Property CategoryName As String

        Public Property SalesPersonName As String
    End Class

    Public Class ProductDataRecord

        Public Property ProductName As String

        Public Property CategoryName As String

        Public Property UnitPrice As Decimal
    End Class
End Namespace

File: DevExpress-Demo/CS/PivotGridDemo.Wpf/Helpers/ServerMode/OrderDataContext.cs
using DevExpress.Xpf.DemoBase;
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.IO;
using System.Linq;
using System.Text;

namespace PivotGridDemo {
    public class OrderDataContext : DbContext {
        public static readonly string FileName = Path.GetFullPath("OrderData.db");
        public OrderDataContext()
            : base(SQLiteDataBaseGenerator.CreateConnection(ServerModeViewModel.DatabaseInfo), true) {
        }
        public DbSet<OrderDataRecord> OrderDataRecords { get; set; }
    }

    public class OrderDataRecord {
        public int Id { get; set; }
        public DateTime OrderDate { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public string CustomerName { get; set; }
        public string ProductName { get; set; }
        public string CategoryName { get; set; }
        public string SalesPersonName { get; set; }
    }
    public class ProductDataRecord {
        public string ProductName { get; set; }
        public string CategoryName { get; set; }
        public decimal UnitPrice { get; set; }
    }
}
